### ISSUE TITLE
Stratosphere Developer Overview Images

### ISSUE THREAD

**Issue by markus-h** [stratosphere/stratosphere#496]
I'm currently trying to create some overview images for developers to give a quick initial overview of how different components of Stratosphere are working together.
The aim is to make it easier for new developers to get a general understanding of the overall architecture of the whole project and of single components.

I found it quite hard to find a good way of presentation, that's why I would appreciate some comments and suggestions.
My initial try can be found here: https://github.com/stratosphere/stratosphere/wiki/DeveloperOverview

**Comment by StephanEwen** [comment]
Nice start. There are two comments:
1. The local executor is not a part of the compile/execute chain. It is
   rather one every point. I would remove it from the picture.
2. The union replacer occurs twice.

Looks good otherwise.
Am 24.02.2014 13:52 schrieb "markus-h" notifications@github.com:

**Comment by markus-h** [comment]
Thanks for your feedback! I adjusted the images accordingly.

**Comment by MoeweX** [comment]
Hi, are you still working on it? I think this is a great idea because Stratosphere is a little bit overwhelming at the beginning. If not this issue can be closed ... .

**Comment by markus-h** [comment]
I had a discussion on this issue with some other developers a few days ago and we agreed that it would be really nice to have a more or less complete development overview of the system for new developers. There will be some efforts torwards documentation during the next weeks, so perhaps we should keep this issue open for now.

**Comment by MoeweX** [comment]
I would appreciate something like this.

**Comment by fhueske** [comment]
Yes, I definitely agree that we need (better) documentation of the projects internals to help new contributors to get familiar with the codebase. 

I think we need a separate section on the website for this purpose.
What do you think?

**Comment by StephanEwen** [comment]
I would suggest to use the github wiki for that. Also easier to extend and
update. We can link to it from the website.

**Comment by fhueske** [comment]
Yes, I agree. The Github wiki is the right place to put this information.

**Comment by uce** [comment]
+1

It's also possible to generate the website docs from the GitHub wiki. I know that netty.io is doing something along those lines. I will look into it later.
